api-reference/10 UI Components/CollectionWidget/1 Configuration/noDataText.md

```diff
@@ -8,3 +8,4 @@ default: 'No data to display'
 The text or HTML markup displayed by the UI component if the item collection is empty.
 
 ---
+The {WidgetName} component evaluates this property's value. This makes {WidgetName} potentially vulnerable to XSS attacks. To guard against them, encode the HTML markup before you assign it to this property. Refer to the following help topic for more information: [Potentially Vulnerable API - noDataText](/Documentation/Guide/Common/Security_Considerations/#HTML_Encoding/Potentially_Vulnerable_API/noDataText).
```

Comment:
Hi Armina,

This description appears in a help topic dedicated to a particular property. This topic looks as follows:
![изображение](https://user-images.githubusercontent.com/18551316/126751379-c9dfbcbd-ce50-40bc-a122-ab9ca6847f3a.png)
As you can see, there's no ambiguity as to which property we are talking about, because its name is in the header, and the whole article is about it.

> The {WidgetName} component evaluates this property's value. This **[what this?]** makes {WidgetName} potentially vulnerable to XSS attacks.

"This" referres to the fact that the component evaluates the property's value. If there wouldn't be a guideline against "which" clauses, the sentence could be "The {WidgetName} component evaluates this property's value, which makes {WidgetName} potentially vulnerable to XSS attacks."

> To guard against them, encode the HTML markup before you assign it **[it what?]** to this property.

This sentence contains only one object that can be referred to as "it" - the markup.